src/vendor/processors/stripe.py

```diff
@@ -643,16 +643,29 @@ def build_setup_intent(self, payment_method_id):
     
     def build_subscription(self, subscription, payment_method_id):
         price = subscription.offer.get_current_price_instance()
+        sub_discount = 0
+        promotion_code = None
 
-        stripe_base_fee = self.get_stripe_base_fee_amount(self.invoice.total)
-        stripe_recurring_fee = self.get_recurring_fee_amount(self.invoice.total)
-        application_fee = self.get_application_fee_amount(self.invoice.total)
+        # Stripe Fees are not adjusted by the discount until discount duration has been implemented
+        stripe_base_fee = self.get_stripe_base_fee_amount(price.cost)
+        stripe_recurring_fee = self.get_recurring_fee_amount(price.cost)
+        application_fee = self.get_application_fee_amount(price.cost)
 
-        total_fee_percentage = self.calculate_fee_percentage(self.invoice.total, stripe_base_fee + stripe_recurring_fee + application_fee)
+        if hasattr(self.invoice, 'coupon_code') and self.invoice.coupon_code.count():
```

Comment:
might be able to simplify by changing this to `coupon_code := self.invoice.coupon_code.first():` and removing the following line?